docker_cli/commands.py

```diff
@@ -0,0 +1,143 @@
+import os
+from shlex import split
+from subprocess import DEVNULL, PIPE, CalledProcessError, run
+from typing import Iterable, List, Union
+
+from ._docker_mamba import mamba_lockfile_command
+from ._exceptions import ImageNotFoundError
+from ._image import Image
+from ._utils import is_conda_pkg_name, universal_tag_prefix
+
+
+def dropin(tag: str) -> None:
+    """
+    Initiates a drop-in session on an image.
+
+    Parameters
+    ----------
+    tag : str
+        The tag or ID of the image.
+    """
+    try:
+        image: Image = Image(tag)
+    except ImageNotFoundError as err:
+        print(f"Error: {str(err)}")
+        exit(1)
```

Comment:
Rather than catching the exception and calling `exit()` directly, I think it'd be better to just let the exception go uncaught. If an exception is uncaught it will also cause the program to abort with a non-zero error code (just like `exit(1)`), but exceptions are more powerful in general since they can preserve the stack trace and allow the calling code to catch the exception and handle it differently.

Apart from that, there are a couple of things happening here that I'm not a fan of:

* We're losing information about the exception type and just printing "Error" following by the exception message.
* We're printing an error message to stdout instead of stderr

So I think the simple thing of just leaving the exception uncaught is also the best thing to do in this case.

```suggestion
    image: Image = Image(tag)
```